lib/api/controllers/funnelController.js

```diff
@@ -200,16 +201,19 @@ class FunnelController {
       return -1;
     }
 
-    let result = 0;
+    // if the connection is closed there is no need to execute the request
+    // => discarding it
+    if (!this.kuzzle.router.isConnectionAlive(request.context)) {
+      delete this.pendingRequests[request.id];
+      return 0;
+    }
+
     this.checkRights(request)
       .then(modifiedRequest => this.processRequest(modifiedRequest))
       .then(processResult => callback(null, processResult))
-      .catch(err => {
-        this._executeError(err, request, true, callback);
-        result = 1;
-      });
+      .catch(err => this._executeError(err, request, true, callback));
```

Comment:
Your proposal about returning a promise is a no-go.
But I agree with updating the jsdoc.